tictactoe.js

```diff
@@ -0,0 +1,144 @@
+const readline = require("readline");
+
+const readlineInterface = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+// Ask question
+const ask = questionText => {
+  return new Promise((resolve, reject) => {
+    readlineInterface.question(questionText, resolve);
+  });
+};
+
+let values = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
```

Comment:
Technically these are not values, but cell indices, right?

You've decided to use the cell number (or index) as a value for empty cells. That's workable but it does muddy the waters, mixing up the *data model* ("empty cell") with the *presentation* ("grid position 5" or "press 5 to move here"). Consider using only 3 values in each cell (e.g. 'X' / 'O' / null) and then making the `grid` function responsible for figuring out what to display based on the value.